libsolidity/CMakeLists.txt

```diff
@@ -29,7 +29,16 @@ add_library(solidity ${sources} ${headers})
 target_link_libraries(solidity PUBLIC evmasm devcore ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY})
 
 if (${Z3_FOUND})
-  target_link_libraries(solidity PUBLIC ${Z3_LIBRARY})
+  if(SOLC_LINK_STATIC)
```

Comment:
I think this changeset could be merge in an individual PR with a changelog entry ("z3 can now be statically linked"). This doesn't depend on this PR having circleci caching.